src/model/clients/HttpClient.js

```diff
@@ -112,21 +112,15 @@ export class HttpClient {
       body: data,
     } = this.requestConfig;
 
-    try {
-      await this.onBeforeRequest(this.requestConfig);
-
-      const response = await axios({
-        baseURL,
-        timeout,
-        headers,
-        url,
-        method,
-        data,
-      });
-
-      return response.data;
-    } catch (err) {
-      return Promise.reject(err.response.data);
-    }
+    await this.onBeforeRequest(this.requestConfig);
+
+    return axios({
```

Comment:
By removing the try/catch block, it means that this promise will resolve/reject with objects created by `axios`, which means that every file which uses `HttpClient` or any class that extends it will depend on `axios`.

when `axios()` resolves, it resolves with an object of type `AxiosResponse` [(see type)](https://github.com/axios/axios/blob/521444513969a08ec5ef943c41ba0812845ed4f9/index.d.ts#L73), and it rejects with an object of type `AxiosError` [(see type)](https://github.com/axios/axios/blob/521444513969a08ec5ef943c41ba0812845ed4f9/index.d.ts#L82).

The original's try/catch block's purpose is to resolve/reject the promise with the response that we get from the server, so no files will actually depend on `axios`, except for this one.

This is a breaking change, the try/catch block is needed in the current implementation.

Reply:
Ok, I see... then we could do the same you were doing but without returning another resolved promise

Reply:
What do you mean? Can you please provide an example?

Reply:
Pushed already